### PR TITLE
Cloud and Server section refactor

### PR DESCRIPTION
**Pull request by billpalombi**
These changes:
- ~Renames the "Cloud and Server" section to "Manage Prefect." feedback welcome on whether thats the best name~ Kept it the same
- Groups all of the pages that are specific to Cloud together
- Renames some pages for brevity and clarity

### CODE REVIEW

**Comment by discdiver** [comment]
"Manage Prefect" - I likethe idea of using a verb, but fear that one might invite more confusion than leaving the title as is.

File: docs/mint.json [hunk 1]
```diff
@@ -174,31 +174,38 @@
       ]
     },
     {
-      "group": "Cloud and Server",
+      "group": "Manage Prefect",
```
Comment (billpalombi) [inline docs/mint.json:177]:
```suggestion
      "group": "Cloud and Server",
```

**Comment by billpalombi** [comment]
You're right, updated to remove that change

File: docs/3.0rc/manage/cloud/troubleshoot-cloud.mdx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Troubleshooting for Prefect Cloud
+title: Troubleshooting Prefect Cloud
```
Comment (discdiver) [inline docs/3.0rc/manage/cloud/troubleshoot-cloud.mdx:2]:
```suggestion
title: Troubleshoot Prefect Cloud
```

**Review by discdiver — APPROVED** [review]
One minor suggestion on verb tense for consistency.

**Pull request merged**